src/components/CodeBlock/index.tsx

```diff
@@ -71,24 +75,48 @@ export const CodeBlock = (props: CodeBlockProps) => {
         }
     }
 
+    const copyToClipboard = () => {
+        if (navigator.clipboard) {
+            navigator.clipboard.writeText(code || props.children.props.children.trim())
+            setHasBeenCopied(true)
+            setTooltipVisible(true)
+            setTimeout(() => {
+                setTooltipVisible(false)
+            }, 1000)
+        }
+    }
+
     return (
-        <Highlight
-            {...defaultProps}
-            code={code || props.children.props.children.trim()}
-            language={language as Language}
-            theme={theme}
-        >
-            {({ className, style, tokens, getLineProps, getTokenProps }) => (
-                <pre className={className} style={{ ...style, padding: '20px' }} id={codeBlockId}>
-                    {tokens.map((line, i) => (
-                        <div key={i} {...getLineProps({ line, key: i })}>
-                            {line.map((token, key) => (
-                                <span key={key} {...getTokenProps({ token, key })} />
-                            ))}
-                        </div>
-                    ))}
-                </pre>
-            )}
-        </Highlight>
+        <>
+            <Highlight
+                {...defaultProps}
+                code={code || props.children.props.children.trim()}
+                language={language as Language}
+                theme={theme}
+            >
+                {({ className, style, tokens, getLineProps, getTokenProps }) => (
+                    <pre className={className} style={{ ...style, padding: '20px' }} id={codeBlockId}>
+                        <Tooltip title="Copied!" visible={tooltipVisible}>
```

Comment:
There just may be confusion if `navigator.clipboard` is not available, copy button shouldn't be shown then (e.g. make `copyToClipboard` `null` then, and only show `CopyOutlined` if `copyToClipboard` is not `null`).

Reply:
ah yes that's true!